scripts/ci/gitlab/pipeline/build.yml

```diff
@@ -79,7 +79,7 @@ build-malus:
     - .compiler-info
     - .collect-artifacts
   script:
-    - time cargo build --locked --profile testnet --verbose -p polkadot-test-malus
+    - time cargo build --locked --profile testnet --verbose -p polkadot-test-malus -p polkadot-execute-worker -p polkadot-prepare-worker
```

Comment:
I tried different approaches and ended up explicitly building workers for `malus`.

[This one](https://gist.github.com/s0me0ne-unkn0wn/bbd83fe32ce10327086adbf13e750eec) even works but has some flaws and looks to hacky to end up in the production code.

Any better ideas are appreciated.

Reply:
Is it possible to have a separate workspace for `malus`, then build it the exact same way we do polkadot? i.e. if we list the workers as default-members, we don't have to specify them with `-p` for each build command.

Reply:
> Is it possible to have a separate workspace for `malus`

Unfortunately, no, nested workspaces are proposed but not supported yet. I went through a lot of interesting proposals while I was researching possible solutions, like intersecting workspaces and binary dependencies, but neither of them is implemented now.

Reply:
> I went through a lot of interesting proposals while I was researching possible solutions ... but neither of them is implemented now.

Yeah I know the feeling. I think `-p ..` is fine, maybe just update the readme with build instructions like we did for node metrics.

Reply:
I've found out that `cargo test --workspace --profile testnet` triggers building some auxiliary binaries like `adder-collator` and `undying`, which are required for tests residing in `tests/`. But I still need to understand what mechanism triggers those builds. Probably it can be used for `malus` as well.